Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -369,51 +414,58 @@ Vector<uint8_t> CryptoKeyEC::platformExportPkcs8() const
     size_t expectedKeySize = keySizeInBytes * 3 + 1; // 04 + X + Y + D
     Vector<uint8_t> keyBytes(expectedKeySize);
     size_t keySize = keyBytes.size();
-    if (UNLIKELY(CCECCryptorExportKey(kCCImportKeyBinary, keyBytes.data(), &keySize, ccECKeyPrivate, m_platformKey.get()) || keySize != expectedKeySize))
-        return { };
+    Vector<uint8_t> result;
+    return WTF::switchOn(m_platformKey, [&](const std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>& options) mutable ->Vector<uint8_t> {
+
+        if (UNLIKELY(CCECCryptorExportKey(kCCImportKeyBinary, keyBytes.data(), &keySize, ccECKeyPrivate, options.get()) || keySize != expectedKeySize))
+            return { };
+
+        const uint8_t* oid;
+        size_t oidSize = getOID(namedCurve(), oid);
+
+        // InitialOctet + 04 + X + Y
+        size_t publicKeySize = keySizeInBytes * 2 + 2;
+        // BIT STRING + length(?) + publicKeySize
+        size_t taggedTypeSize = bytesNeededForEncodedLength(publicKeySize) + publicKeySize + 1;
+        // VERSION + OCTET STRING + length(1) + private key + TaggedType1(1) + length(?) + BIT STRING + length(?) + publicKeySize
+        size_t ecPrivateKeySize = sizeof(Version) + keySizeInBytes + bytesNeededForEncodedLength(taggedTypeSize) + bytesNeededForEncodedLength(publicKeySize) + publicKeySize + 4;
+        // SEQUENCE + length(?) + ecPrivateKeySize
+        size_t privateKeySize = bytesNeededForEncodedLength(ecPrivateKeySize) + ecPrivateKeySize + 1;
+        // VERSION + SEQUENCE + length(1) + OID id-ecPublicKey + OID secp256r1/OID secp384r1/OID secp521r1 + OCTET STRING + length(?) + privateKeySize
+        size_t totalSize = sizeof(Version) + sizeof(IdEcPublicKey) + oidSize + bytesNeededForEncodedLength(privateKeySize) + privateKeySize + 3;
+        
+        result.reserveInitialCapacity(totalSize + bytesNeededForEncodedLength(totalSize) + 1);
+        result.append(SequenceMark);
```

Comment:
Ditto with reserveInitialCapacity and uncheckedAppend(span<uint8_t>).

Since this is preexisting code this is probably fine to keep as is but we could do this as a follow-up

Reply:
ok